sdk/openai/openai/src/OpenAIClient.ts

```diff
@@ -119,9 +119,21 @@ export class OpenAIClient {
     let cred: KeyCredential | TokenCredential;
     if (isCred(credOrOptions)) {
       endpoint = endpointOrOpenAiKey as string;
-      cred = credOrOptions;
-      opts = options;
-      this._isAzure = true;
+      this._isAzure = /(azure\.com|endpoint)[/]?$/.test(endpoint);
+      if (this._isAzure) {
+        cred = credOrOptions;
+        opts = options;
+      } else {
+        cred = credOrOptions as KeyCredential;
+        const { credentials, ...restOpts } = options;
+        opts = {
```

Comment:
It's great that we're adding this! Especially given OpenAI's API is becoming a bit of a "de facto" cross-LLM surface, it feels like the custom endpoint questions keep piling up.

Can we add a (very brief) readme-facing example at the same time? Having one of those to point customers directly towards typically pays for itself *very* quickly.